astropy/coordinates/sky_coordinate.py

```diff
@@ -1317,20 +1317,20 @@ def search_around_sky(self, searcharoundcoords, seplimit):
 
         Parameters
         ----------
-        searcharoundcoords : `~astropy.coordinates.SkyCoord` or `~astropy.coordinates.BaseCoordinateFrame`
+        searcharoundcoords : coordinate-like
             The coordinates to search around to try to find matching points in
             this `SkyCoord`. This should be an object with array coordinates,
             not a scalar coordinate object.
-        seplimit : `~astropy.units.Quantity` with angle units
-            The on-sky separation to search within.
+        seplimit : `~astropy.units.Quantity`
+            The on-sky separation to search within. Must have angular units.
```

Comment:
Very out of scope for this PR, but this is reminding me that in gala, I use the convention:
```python
argname : type [physical type]
```
if the type is quantity-like. 

In any case, maybe this should be:
```suggestion
        seplimit : angle-like
            The on-sky separation to search within.
```
?

Reply:
"angle-like" includes str, which is not currently supported.